src/payment/models/dex/strategies/purchase-liquidity/purchase-stock-liquidity.strategy.ts

```diff
@@ -0,0 +1,103 @@
+import { Injectable } from '@nestjs/common';
+import { LiquidityOrder } from '../../entities/liquidity-order.entity';
+import { LiquidityOrderRepository } from '../../repositories/liquidity-order.repository';
+import { LiquidityService } from '../../services/liquidity.service';
+import { PurchaseLiquidityStrategy } from './purchase-liquidity.strategy';
+import { MailService } from 'src/shared/services/mail.service';
+import { LiquidityOrderFactory } from '../../factories/liquidity-order.factory';
+import { LiquidityRequest } from '../../services/dex.service';
+import { AssetCategory } from 'src/shared/models/asset/asset.entity';
+import { NotEnoughLiquidityException } from '../../exceptions/not-enough-liquidity.exception';
+
+@Injectable()
+export class PurchaseStockLiquidityStrategy extends PurchaseLiquidityStrategy {
```

Comment:
I see a lot of code duplicated in `PurchaseStockLiquidityStrategy` and `PurchaseCryptoLiquidityStrategy` (everything except getting swap asset/amount). Did you think about merging them? It could be a could idea, especially if we might at some time also have multiple swap assets for crypto tokens.

Reply:
in my opinion it's fine, I would still not merge them, the key difference is in getSuitableSwapAsset() in case of crypto, the rest is more orchestration methods. If I would be too strict I would create super class or even extract some share "util"... but in this case the game doesn't worth the candles in my opinion, will just make more complicated without big benefit imho

Reply:
as discussed, will rework ;)